python/tskit/genotypes.py

```diff
@@ -102,21 +138,40 @@ def alleles(self) -> tuple:
         """
         return self._ll_variant.alleles
 
+    @property
+    def samples(self) -> tuple:
+        """
+        A numpy array of the node ids whose genotypes will be returned
+        by the :meth:`genotypes` method.
+        """
+        return self._ll_variant.samples
+
     @property
     def genotypes(self) -> np.ndarray:
         """
         An array of indexes into the list ``alleles``, giving the
         state of each sample at the current site.
         """
+        self._check_decoded()
         return self._ll_variant.genotypes
 
+    @property
+    def isolated_as_missing(self) -> bool:
+        """
+        True if isolated samples are decoded to missing data. If False, isolated
+        samples are decoded to the ancestral state.
+        """
+        return self._ll_variant.isolated_as_missing
+
     @property
     def has_missing_data(self) -> bool:
         """
         True if there is missing data for any of the
         samples at the current site.
         """
-        return self._ll_variant.alleles[-1] is None
+        return (
```

Comment:
This is creating the ``alleles`` tuple twice I guess? This gets called a lot, so makes sense to keep it tight.